static_frame/test/unit/test_type_blocks.py

```diff
@@ -4030,6 +4031,78 @@ def test_unified_dtypes_c(self) -> None:
         a1 = np.array([False, True, False])
         tb1 = TypeBlocks.from_blocks((a1, ))
         self.assertTrue(tb1.unified_dtypes)
+    
+    #---------------------------------------------------------------------------
+    def test_sizeof_a(self) -> None:
+        a = np.array([1, 2, 3])
+        tb = TypeBlocks.from_blocks(a)
+        self.assertTrue(getsizeof(tb), sum(getsizeof(e) for e in [
+            # _blocks
+            np.array([1, 2, 3]),
+            [np.array([1, 2, 3])],
+            # _index
+            0,
+            (0, 0),
+            [(0, 0)],
+            # _shape
+            3, 1,
+            (3, 1),
+            # _dtypes
+            np.dtype('int64'),
+            [np.dtype('int64')],
+            # _row_dtype is np.dtype('int64'), skipped since already included
+            None # to add the size+garbage collector overhead of the TypeBlocks instance itself
+        ]))
+
+    def test_sizeof_b(self) -> None:
+        tb = TypeBlocks.from_blocks([
+            np.array([1, 2, 3]),
+            np.array([4, 5, 6])
+        ])
+        self.assertEqual(getsizeof(tb), sum(getsizeof(e) for e in [
+            # _blocks
+            np.array([1, 2, 3]),
+            np.array([4, 5, 6]),
+            tb._blocks, # [np.array([1, 2, 3]), np.array([4, 5, 6])],
```

Comment:
The `getsizeof` size of `_blocks` is not equal to the size of the constructed list. from_blocks uses append to create the _blocks array. Using append triggers allocation for a few more elements. I used references to the member variables rather than constructing these manually to mitigate this inconsistency.
```
>>> a = []
>>> a.append(2)
>>> b = [2]
>>> getsizeof(a)
88
>>> getsizeof(b)
64
```

Reply:
That is an interesting observation. I think our goal here is to find memory foot print of SF containers. If, for example, we want to multiprocess a large `Frame` we would need to know how much memory we might need in each process. That would include over-allocation from mutable containers. So maybe it is okay (better?) to included those over-allocated positions. You could also make it an option into your utility function.

Reply:
I think it's better to include the over-allocated positions in the default results.

Are you suggesting to include an option to compute the "minimal allocation" for mutable containers like lists in the utility function? I am not sure that such a statistic would be useful unless we are looking for places to optimize (such as replacing mutable containers with minimally allocated ones) or launching a different thread somehow uses the "minimal allocation"

Reply:
Great, I agree: lets count the over-allocated positions. I do not see a strong need to implement a "minimal allocation" option: I just thought you were implying it was your preference!  But no problem only doing it one way.